File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> rep;
        if (!root) return rep;
        rep.push_back({root->val});
        auto l_rep = levelOrder(root->left);         
        auto r_rep = levelOrder(root->right);
        auto l = l_rep.size(); auto r = r_rep.size();
        for (int i = 0; i < min(l, r); i++)
            l_rep[i].insert(l_rep[i].end(), r_rep[i].begin(), r_rep[i].end());
        rep.insert(rep.end(), l_rep.begin(), l_rep.end());
        if (r > l) rep.insert(rep.end(), r_rep.begin() + l, r_rep.end());
        return rep;
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        stack = [(root, 0)]
        ans = []
        while stack:
            base, height = stack.pop()
            if base is None: continue
            if len(ans) <= height: ans.append(base.val)
            ltree, rtree = base.left, base.right
            stack.append((ltree, height + 1))
            stack.append((rtree, height + 1))
        return ans